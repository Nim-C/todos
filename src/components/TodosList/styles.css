#message-no-todos {
  line-height: 54px;
  text-align: center;
}
