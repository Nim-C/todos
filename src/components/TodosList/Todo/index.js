import React, { useCallback } from 'react'
import classnames from 'classnames'
import './styles.css'

import { ReactComponent as TrashCanIcon } from '../../../trash-solid.svg'
import { useDispatch } from 'react-redux'
import { removeTodo, toggleTodo } from '../../../features/todos/todosSlice'

const Todo = ({ id, name, completed }) => {
  const dispatch = useDispatch()

  const handleRemoveTodo = useCallback(() => {
    dispatch(removeTodo({ id }))
  }, [dispatch, id])

  const handleToggleTodo = useCallback(() => {
    dispatch(toggleTodo({ id }))
  }, [dispatch, id])

  return (
    <div
      className={classnames('card', 'rounded', 'bg-white', 'mb-4', 'col-12', {
        shadow: !completed,
        'shadow-sm': completed
      })}
    >
      <div
        className={classnames(
          'card-body',
          'd-flex',
          'justify-content-between',
          'todo-name'
        )}
      >
        <div className={classnames('custom-control', 'custom-checkbox')}>
          <input
            type="checkbox"
            className="custom-control-input"
            id={`todo-${id}-checkbox`}
            onChange={handleToggleTodo}
          />
          <label
            className={classnames('custom-control-label', {
              'todo-completed': completed
            })}
            htmlFor={`todo-${id}-checkbox`}
          >
            {name}
          </label>
        </div>
        <TrashCanIcon className="trash-icon" onClick={handleRemoveTodo} />
      </div>
    </div>
  )
}

export default Todo
