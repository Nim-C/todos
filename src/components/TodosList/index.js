import React from 'react'
import TodosForm from '../TodosForm'
import classnames from 'classnames'
import Todo from './Todo'
import Either from '../util-components/Either'
import './styles.css'
import { useSelector } from 'react-redux'
import { selectTodosList } from '../../features/todos/selectors'

const TodosList = () => {
  const list = useSelector(selectTodosList)
  const todos = Object.keys(list).map(todoKey => {
    const currTodo = list[todoKey]
    return <Todo {...currTodo} key={todoKey} />
  })

  return (
    <div
      className={classnames(
        'd-flex',
        'flex-column',
        'align-items-center',
        'mx-auto',
        'col-12',
        'col-lg-4'
      )}
    >
      <TodosForm />
      <div className={classnames('todos-list-container', 'w-100')}>
        <Either
          condition={Object.keys(list).length}
          ifTrue={todos}
          ifFalse={<h4 id="message-no-todos">No Todoâ€™s to show...</h4>}
        />
      </div>
    </div>
  )
}

export default TodosList
