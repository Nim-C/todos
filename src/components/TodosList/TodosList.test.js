import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import store from '../../store'

import TodosList from './'

test('renders Todos List without failing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <TodosList />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
