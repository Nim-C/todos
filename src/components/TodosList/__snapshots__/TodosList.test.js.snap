// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Todos List without failing 1`] = `
<div
  className="d-flex flex-column align-items-center mx-auto col-12 col-lg-4"
>
  <div
    className="d-flex justify-content-between p-0 col-12 form-container mb-4"
  >
    <input
      className="todo-name-input form-control mr-3 my-2"
      onChange={[Function]}
      placeholder="Add todo’s..."
      type="text"
      value=""
    />
    <button
      className="btn btn-success my-2"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
  <div
    className="todos-list-container w-100"
  >
    <h4
      id="message-no-todos"
    >
      No Todo’s to show...
    </h4>
  </div>
</div>
`;
